File: musicShopProject/Model/Orders/OrderItemBlank.cs
namespace musicShopProject.Model.Orders;

public partial class OrderItemBlank
{
    public Guid? Id { get; set; }
    public Guid? OrderId { get; set; }
    public Guid? ProductId { get; set; }
    public Decimal? ProductPrice { get; set; }
    public Guid? ProductCategoryId { get; set; }

}

public partial class OrderItemBlank
{
    public class Validated
    {
        public Guid Id { get; }
        public Guid OrderId { get; }
        public Guid ProductId { get; }
        public Decimal ProductPrice { get; }
        public Guid ProductCategoryId { get; }

        public Validated(Guid id, Guid orderId, Guid productId, Decimal productPrice, Guid productCategoryId)
        {
            Id = id;
            OrderId = orderId;
            ProductId = productId;
            ProductPrice = productPrice;
            ProductCategoryId = productCategoryId;
        }
    }
}

File: musicShopProject/tentashop/src/domain/orders/orderItemBLank.ts
import { OrderItem } from "./orderItem";

export interface OrderItemBlank{
    id: string | null,
    OrderId: string | null,
    ProductId : string | null,
    ProductPrice : number | null,
    ProductCategoryId : string | null
}

export namespace ProductBlank {
    export function getEmpty(): OrderItemBlank {
        return {
            id: null,
            OrderId: null,
            ProductId :  null,
            ProductPrice : null,
            ProductCategoryId : null
        };
    }

    export function toProductBlank(orderItem: OrderItem): OrderItemBlank {
        return {
            id: orderItem.id,
            OrderId: orderItem.orderId,
            ProductId : orderItem.product.id,
            ProductPrice : orderItem.productPrice,
            ProductCategoryId : orderItem.productCategoryId
        };
    }
 }